src/detect-content.c

```diff
@@ -369,6 +381,8 @@ int DetectContentSetup(DetectEngineCtx *de_ctx, Signature *s, const char *conten
 
 error:
     DetectContentFree(de_ctx, cd);
+    if (warning)
+        return 0;
```

Comment:
will this not load the rule, but minus the broken content condition?

so effectively turning
`alert ... content:"|aa a aa|"; dsize:10;`
into
`alert ... dsize:10;`
?

Reply:
Looks like we should either return -2 or -3 here (not sure which though):
```c
    int ret = SigParse(de_ctx, sig, sigstr, dir, &parser);                                               
    if (ret == -3) {                                                                                     
        de_ctx->sigerror_silent = true;                                                                  
        de_ctx->sigerror_ok = true;                                                                      
        goto error;                                                                                      
    }                                                                                                    
    else if (ret == -2) {                                                                                
        de_ctx->sigerror_silent = true;                                                                  
        goto error;                                                                                      
    } else if (ret < 0) {                                                                                
        goto error;                                                                                      
    }                                                                                                    
```

Reply:
Actually, its a -4...  New.. Log the error, but don't error out.  An option we don't have it.  Which handles the previous case as well.

ie)
```
        de_ctx->sigerror_ok = true;                                                                      
```